File: haskell/problem5.hs
-- Problem5.hs
module Problem5 (debePagarImpuestos) where

umbral :: Int
umbral = 53206000  -- COP

-- Funcion que determina si una persona debe pagar impuestos
debePagarImpuestos :: Int -> Int -> Bool
debePagarImpuestos edad ingresos =
    edad >= 18 && ingresos >= umbral

-- funcion principal de prueba
main :: IO ()
main = do
    print (debePagarImpuestos 25 60000000) -- True
    print (debePagarImpuestos 16 80000000) -- False
    print (debePagarImpuestos 30 50000000) -- False
    print (debePagarImpuestos 18 53206000) -- True

File: python/problem5.py
# problem5.py
IMPUESTOS = 53206000  

def debe_pagar_impuestos(edad: int, ingresos: float) -> bool:
    """
    Determina si un usuario debe pagar impuestos.
    """
    if edad < 0 or edad > 120:
        raise ValueError("Edad es valiida.")
    if ingresos < 0:
        raise ValueError("Ingresos no pueden ser negativos.")
    return edad >= 18 and ingresos >= IMPUESTOS


if __name__ == "__main__":
    try:
        edad = int(input("Ingrese su edad: "))
        ingresos = float(input("Ingrese el total acumulado de consignaciones anuales: "))

        if debe_pagar_impuestos(edad, ingresos):
            print("Debe pagar impuestos.")
        else:
            print("No debe pagar impuestos.")
    except ValueError as e:
        print("Error:", e)

